### PR TITLE
Move retrieval action to class pattern

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

This PR moves the last remaining action to the runner class pattern.

### CODE REVIEW

File: front/lib/api/assistant/actions/retrieval.ts [hunk 1]
```diff
@@ -244,6 +242,455 @@ export class RetrievalAction extends BaseAction {
...
+        functionCallId: action.functionCallId,
+        functionCallName: action.functionCallName,
+        documents,
+        step: action.step,
+      }),
+    };
+  }
+}
+
 export function retrievalAutoQueryInputSpecification() {
```
Comment (PopDaph) [inline front/lib/api/assistant/actions/retrieval.ts:694]: No changes in this one and the following ones right? (asking cause I did not review them, assuming it's only moved code)

File: front/lib/api/assistant/legacy_agent.ts [hunk 2]
```diff
@@ -376,15 +363,20 @@ async function* runAction(
   }
 
   if (isRetrievalConfiguration(action)) {
```
Comment (PopDaph) [inline front/lib/api/assistant/legacy_agent.ts:365]: Goal is still to get rid of this right? (Assuming we move the code handling the events and other custom cases.

**Review by PopDaph — APPROVED** [review]
👍🏻

File: front/lib/api/assistant/actions/retrieval.ts (hunk 1 above)
Comment (flvndvd) [reply front/lib/api/assistant/actions/retrieval.ts:694]: Yep correct.

File: front/lib/api/assistant/legacy_agent.ts (hunk 2 above)
Comment (flvndvd) [reply front/lib/api/assistant/legacy_agent.ts:365]: Yep last step is to move the streaming events to their respective classes.

**Pull request merged**